tests/server_notices/test_resource_limits_server_notices.py

```diff
@@ -55,25 +55,18 @@ def prepare(self, reactor, clock, hs):
         self._rlsn._store.user_last_seen_monthly_active = Mock(
             return_value=defer.succeed(1000)
         )
-        self._send_notice = self._rlsn._server_notices_manager.send_notice
-        self._rlsn._server_notices_manager.send_notice = Mock()
-        self._rlsn._state.get_current_state = Mock(return_value=defer.succeed(None))
-        self._rlsn._store.get_events = Mock(return_value=defer.succeed({}))
-
+        self._rlsn._server_notices_manager.send_notice = Mock(
+            return_value=defer.succeed(Mock())
+        )
         self._send_notice = self._rlsn._server_notices_manager.send_notice
 
         self.hs.config.limit_usage_by_mau = True
         self.user_id = "@user_id:test"
 
-        # self.server_notices_mxid = "@server:test"
-        # self.server_notices_mxid_display_name = None
-        # self.server_notices_mxid_avatar_url = None
-        # self.server_notices_room_name = "Server Notices"
-
         self._rlsn._server_notices_manager.get_or_create_notice_room_for_user = Mock(
-            returnValue=""
+            return_value=defer.succeed("!something:localhost")
```

Comment:
I believe the original `returnValue` was a typo of `return_value` and thus didn't do anything. The relevant code that failed after async/await conversion was:

https://github.com/matrix-org/synapse/blob/bf700782fae0879f2e8d08fcca7b69e23cd7d892/synapse/server_notices/resource_limits_server_notices.py#L82-L87

`Mock`s by default return `Mock`s, so the code would end up as `room_id = yield Mock()`. `yield` would give up here and just return the `Mock`, thus `room_id` would be a Mock, which is not the intention of the test, but allowed it to pass, as a `Mock` is truthy, and thus the `if not room_id` condition would fail.

In contrast, `await` will error here stating that a `Mock` is not awaitable. No worries, I just fixed the `return_value` typo. Now `room_id` was `""`, but this failed the `if not room_id` condition now as `""` is falsey.

So, I switch it to a fake room id and here we are :slightly_smiling_face:

Reply:
Ah-ha! Nice. Thanks for the explanation!